src/sql/azdata.proposed.d.ts

```diff
@@ -2073,6 +2073,382 @@ declare module 'azdata' {
+		 * Bracketed portions are optional for alpha/transparency.
+		 */
+		backgroundColor: string;
+		/**
+		 * Border color for chart elements.  May be a name ('red'), hex ('#FFFFFF[77]), or RGB ('rgb(255, 255, 255[, 0.5])).
+		 * Bracketed portions are optional for alpha/transparency.
+		 */
+		borderColor?: string;
+	}
+
+	export interface ChartDataEntry extends ChartDataEntryBase {
+		/**
+		 * Value of one-dimensional data point
+		 */
+		value: Chart1DPoint | number;
+	}
+
+	export interface ChartDataSet<TVal extends Chart1DPoint | number> extends ChartDataEntryBase {
+		data: TVal[];
+	}
+
+	/**
+	 * One-dimensional data point
+	 */
+	export interface Chart1DPoint {
+		/**
+		 * Value for a one-dimensional data point, or the x-coordinate for a multi-dimensional data point
+		 */
+		x: number;
+	}
+
+	/**
+	 * Two-dimensional data point
+	 */
+	export interface Chart2DPoint extends Chart1DPoint {
+		/**
+		 * Y-coordiate for a multi-dimensional data point
+		 */
+		y: number;
+	}
+
+	/**
+	 * Three-dimensional data point
+	 */
+	export interface Chart3DPoint extends Chart2DPoint {
+		/**
+		 * Radius for a bubble data point, in pixels
+		 */
+		r: number;
+	}
+
+	//#endregion
+
+	//#region Chart general option types
+
+	/**
+	 * Base options for a chart
+	 */
+	export interface ChartOptions<TChartType extends ChartType> {
+		/**
+		 * Never needs to be set or used.  Only present for the TypeScript compiler to recognize the pairing between same-chart Data and Options types.
+		 */
+		// DevNote:
+		// This works because it gets compiled to (e.g.) `'bar' | undefined, forcing it to be associated with BarChartData
+		// and preventing it from being paired with PieChartData.
+		type?: TChartType;
+
+		/**
+		 * Title of the chart.  Set to `undefined` to not display the title.
+		 */
+		chartTitle?: string;
+
+		/**
+		 * Whether to display the legend.  Defaults to true.
+		 */
+		legendVisible?: boolean;
+
+		/**
+		 * Free-form options that are mixed into the display options for the underlying charting library to support all missing options functionality.
+		 * Chart.js is the current library powering ADS charting, but this is subject to change in the future.
+		 */
+		freeformOptions?: any;
+	}
+
+	/**
+	 * Base options for scales
+	 */
+	export interface ScaleOptions {
+		/**
+		 * Whether to begin the scale at zero
+		 */
+		beginAtZero?: boolean;
+
+		/**
+		 * Minimum value of the scale
+		 */
+		min?: number;
+
+		/**
+		 * Maxium value of the scale
+		 */
+		max?: number;
+
+		/**
+		 * Whether to add extra space between the scale and the chart
+		 */
+		offset?: boolean;
+
+		/**
+		 * Whether to stack charted values
+		 */
+		stacked?: boolean;
+	}
+
+	//#endregion
+
+	//#region Chart-specific types
+
+	//#region Bar/Horizontal Bar charts
+
+	export interface BarChartDataSet extends ChartDataSet<Chart1DPoint | number> { }
+
+	export interface BarChartDataBase {
+		datasets: BarChartDataSet[];
```

Comment:
doc comment

Reply:
Done.  I have a slight concern with the options since they're more specific to what chartjs supports (and may not map to Highcharts if we make that change), but maybe that's a "cross that bridge when we get to it" situation.